tests/unit/testsuite/Bolt/Boltpay/Block/Checkout/BoltpayTest.php

```diff
@@ -255,6 +289,401 @@ public function testGetSaveOrderURL()
+     * key has been configured
+     *
+     * @covers ::getPublishableKeyForThisPage
+     * @dataProvider getPublishableKeyForThisPage_withAtLeastOneKeyConfiguredProvider
+     *
+     * @param string|null $multiStepKey             the multi-step publishable key, empty string or null indicates not set
+     * @param string|null $paymentOnlyKey           the payment-only publishable key, empty string or null indicates not set
+     * @param string      $routeName                The magento path route name
+     * @param string      $controllerName           The magento path controller name
+     * @param string      $expectedReturnValue      The key that is expected to be returned by the mock
+     *
+     * @throws Exception if there is a problem in setting up test dependencies
+     */
+    public function getPublishableKeyForThisPage_withAtLeastOneKeyConfigured($multiStepKey, $paymentOnlyKey, $routeName, $controllerName, $expectedReturnValue)
+    {
+        $this->setUp_getPublishableKeyForThisPage($multiStepKey, $paymentOnlyKey, $routeName, $controllerName);
+
+        $this->helperMock->expects($this->never())->method('logException');
+        $this->helperMock->expects($this->never())->method('notifyException');
+
+        $actualMultiStepKey =  $this->app->getStore()->getConfig('payment/boltpay/publishable_key_multipage');
+        $actualPaymentOnlyKey = $this->app->getStore()->getConfig('payment/boltpay/publishable_key_onepage');
+
+        $actualReturnedValue = $this->currentMock->getPublishableKeyForThisPage();
+        $this->assertTrue($actualMultiStepKey || $actualPaymentOnlyKey);
+        $this->assertEquals($multiStepKey, $actualMultiStepKey);
+        $this->assertEquals($paymentOnlyKey, $actualPaymentOnlyKey);
+        $this->assertEquals($expectedReturnValue, $actualReturnedValue);
+    }
+
+    /**
+     * Provides data for a multi-step key, or a payment-only key, or for both publishable keys in the context of
+     * the standard cart page, a custom cart page, the standard checkout page, a standard product page, and any
+     * unexpected page identified by route and controller name for
+     * {@see Bolt_Boltpay_Block_Checkout_BoltpayTest::getPublishableKeyForThisPage_withAtLeastOneKeyConfigured()}
+     *
+     * @return array[] in the format of [$multiStepKey, $paymentOnlyKey, $routeName, $controllerName, $predictedReturnValue]
+     */
+    public function getPublishableKeyForThisPage_withAtLeastOneKeyConfiguredProvider()
+    {
+        return [
+            [
+                # When both keys exist on standard cart page, multi-step is used
+                "multiStepKey" => "multi+payOnly-key-on-standard-cart",
+                "paymentOnlyKey" => "payOnly+multi-key-on-standard-cart",
+                "routeName" => "checkout",
+                "controllerName" => "cart",
+                "predictedReturnValue" => "multi+payOnly-key-on-standard-cart"
```

Comment:
Usually in best practice used key 'expect' or 'expects'

Reply:
Thanks.  I actually wanted to name this the same as I did in the test parameters, which is `expectedReturnValue`.  This, I have updated!